# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/AsmJS.cpp
# Commit: a850ff24c2ae
# Full Hash: a850ff24c2aeaefac447d9e501f6e443f395f844
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 1
# Description:
#   Bug 1561521 - Clean up wasm::ToString and drop wasm::ToCString. r=lth
#   
#   wasm::ToCString is losing relevance as we have more constructed types. This
#   commit removes ToCString and replaces all uses with wasm::ToString.
#   
# ==============================================================================

diff -r 3d1a8283d2f8 -r a850ff24c2ae js/src/wasm/AsmJS.cpp
--- a/js/src/wasm/AsmJS.cpp	Tue Aug 18 00:07:29 2020 +0000
+++ b/js/src/wasm/AsmJS.cpp	Tue Aug 18 00:45:24 2020 +0000
@@ -5859,7 +5859,7 @@
 
   if (f.returnedType() != type) {
     return f.failf(usepn, "%s incompatible with previous return of type %s",
-                   ToCString(type), ToCString(f.returnedType()));
+                   ToString(type).get(), ToString(f.returnedType()).get());
   }
 
   return true;